clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -235,56 +167,56 @@ private void parseContainer(XMLExtendedStreamReader reader, PathAddress subsyste
-                        break;
-                    }
+                    this.parseScheduledThreadPool(ScheduledThreadPoolResourceDefinition.EXPIRATION, reader, address, operations);
+                    break;
                 }
                 case ASYNC_OPERATIONS_THREAD_POOL: {
-                    if (this.schema.since(InfinispanSchema.VERSION_4_0)) {
-                        this.parseThreadPool(ThreadPoolResourceDefinition.ASYNC_OPERATIONS, reader, address, operations);
-                        break;
+                    if (this.schema.since(InfinispanSchema.VERSION_14_0)) {
+                        throw ParseUtils.unexpectedElement(reader);
                     }
+                    ROOT_LOGGER.elementIgnored(element.getLocalName());
+                    ParseUtils.requireNoContent(reader);
+                    break;
                 }
                 case LISTENER_THREAD_POOL: {
-                    if (this.schema.since(InfinispanSchema.VERSION_4_0)) {
-                        this.parseThreadPool(ThreadPoolResourceDefinition.LISTENER, reader, address, operations);
-                        break;
-                    }
+                    this.parseThreadPool(ThreadPoolResourceDefinition.LISTENER, reader, address, operations);
+                    break;
                 }
                 case PERSISTENCE_THREAD_POOL: {
-                    if (this.schema.since(InfinispanSchema.VERSION_4_0)) {
-                        if (this.schema.since(InfinispanSchema.VERSION_7_0) && !this.schema.since(InfinispanSchema.VERSION_10_0)) {
-                            this.parseScheduledThreadPool(ThreadPoolResourceDefinition.PERSISTENCE, reader, address, operations);
-                        } else {
-                            this.parseThreadPool(ThreadPoolResourceDefinition.PERSISTENCE, reader, address, operations);
-                        }
-                        break;
+                    if (this.schema.since(InfinispanSchema.VERSION_14_0)) {
+                        throw ParseUtils.unexpectedElement(reader);
                     }
+                    ROOT_LOGGER.elementIgnored(element.getLocalName());
+                    ParseUtils.requireNoContent(reader);
+                    break;
                 }
                 case REMOTE_COMMAND_THREAD_POOL: {
-                    if (this.schema.since(InfinispanSchema.VERSION_4_0)) {
-                        this.parseThreadPool(ThreadPoolResourceDefinition.REMOTE_COMMAND, reader, address, operations);
-                        break;
+                    if (this.schema.since(InfinispanSchema.VERSION_14_0)) {
+                        throw ParseUtils.unexpectedElement(reader);
                     }
+                    ROOT_LOGGER.elementIgnored(element.getLocalName());
+                    ParseUtils.requireNoContent(reader);
+                    break;
```

Comment:
Let's avoid code duplication here with fallthrough:

```
                case ASYNC_OPERATIONS_THREAD_POOL:
                case STATE_TRANSFER_THREAD_POOL:
                case TRANSPORT_THREAD_POOL:
                case REMOTE_COMMAND_THREAD_POOL:
                case PERSISTENCE_THREAD_POOL: {
                    if (this.schema.since(InfinispanSchema.VERSION_14_0)) {
                        throw ParseUtils.unexpectedElement(reader);
                    }
                    ROOT_LOGGER.elementIgnored(element.getLocalName());
                    ParseUtils.requireNoContent(reader);
                    break;
                }
```

Reply:
Fixed.